src/walk_dirs.rs

```diff
@@ -45,17 +45,24 @@ lazy_static! {
     static ref WORKING_DIR_PATH: PathBuf = current_dir().unwrap();
 }
 
-pub fn discover_fences_and_files(start_path: &str) -> Vec<WalkFileData> {
+pub fn discover_fences_and_files(
+    start_path: &str,
+    ignore_external_fences: bool,
+) -> Vec<WalkFileData> {
     let walk_dir = WalkDirGeneric::<(TagList, WalkFileData)>::new(start_path).process_read_dir(
-        |read_dir_state, children| {
+        move |read_dir_state, children| {
             // Custom filter -- retain only directories and fence.json files
             children.retain(|dir_entry_result| {
                 dir_entry_result
                     .as_ref()
                     .map(|dir_entry| {
                         dir_entry.file_type.is_dir()
                             || match dir_entry.file_name.to_str() {
-                                Some(file_name_str) => should_retain_file(file_name_str),
+                                Some(file_name_str) => {
+                                    !(ignore_external_fences.clone()
```

Comment:
Why are you cloning here when the callback already takes ownership of the bool with move?